src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -354,4 +321,38 @@ namespace OpenMS
       std::max(peak_apex_pos - psm.start_position_at_10, psm.end_position_at_10 - peak_apex_pos);
     return psm;
   }
+
+  template <typename PeakContainerConstIteratorT>
+  double PeakIntegrator::findPosAtPerc_(
+    PeakContainerConstIteratorT it_begin,
+    PeakContainerConstIteratorT it_end,
+    const double peak_height,
+    const double perc,
+    const bool is_left_half
+  ) const
+  {
+    const double goal_intensity = peak_height * perc;
+    PeakContainerConstIteratorT closest = is_left_half ? it_begin : it_end - 1; // TODO assuming enough points here
```

Comment:
I like that you set the default values here and then override them as you search for a more optimal value (if one is available)